subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ParallelDownloadsIntegrationTest.groovy

```diff
@@ -0,0 +1,159 @@
+        println "Max concurrent requests: $maxConcurrentRequests"
+        maxConcurrentRequests > 1
+
+        where:
+        repo << ['maven', 'ivy']
+    }
+
+    @Unroll
+    def "downloads artifacts in parallel with project dependencies using #repo"() {
+        countParallelArtifactDownloads()
+
+        given:
+        multipleFilesToDownloadFrom(repo, 8, 8)
+
+        when:
+        run 'resolve'
+
+        then:
+        noExceptionThrown()
+        println "Max concurrent requests: $maxConcurrentRequests"
+        maxConcurrentRequests > 1
+
+        where:
+        repo << ['maven', 'ivy']
+    }
+
+    private void multipleFilesToDownloadFrom(String repo, int externalDependenciesCount = 8, int projectDependenciesCount = 0) {
+        if (repo in ['ivy', 'maven']) {
+            buildFile << """
+            allprojects {
+                repositories {"""
+            if (repo == 'maven') {
+                buildFile << """
+                    maven { url '$mavenHttpRepo.uri' }
+        """
+            } else {
+                buildFile << """
+                    ivy { url '$ivyHttpRepo.uri' }
+        """
+            }
+            buildFile << """                }
+                configurations {
+                    compile
+                }
+            }
+            
+            dependencies {
+        """
+            externalDependenciesCount.times {
+                if (repo == 'maven') {
+                    def module = mavenHttpRepo.module("test", "test$it", "1.0").publish()
+                    module.pom.expectGet()
+                    module.artifact.expectGet()
+                } else if (repo == 'ivy') {
+                    def module = ivyHttpRepo.module("test", "test$it", "1.0").publish()
+                    module.ivy.expectGet()
+                    module.artifact.expectGet()
+                }
+                buildFile << "               compile 'test:test$it:1.0'\n"
+            }
+            projectDependenciesCount.times {
+                settingsFile << "include 'subproject$it'\n"
+                buildFile << "               compile project('subproject$it')\n"
+                file("subproject$it/build.gradle") << """
+                    apply plugin: 'java-library'
+                """
+                file("subproject$it/src/main/java/com/acme/Foo${it}.java") << """
+                    package com.acme;
+                    public class Foo${it} {}
+                """
+            }
+            buildFile << """
+            }
+            
+            task resolve {
+                dependsOn configurations.compile // force execution of jar tasks for local projects
+                doLast {
+                    println configurations.compile.files
+                }
+            }
+"""
+        } else {
+            throw new IllegalArgumentException("Repository must be one of 'ivy', 'maven'")
+        }
+    }
+
+    private void countParallelArtifactDownloads() {
```

Comment:
The `acceptURI` stuff is here in preparation for parallel download of _metadata_.